src/VisualStudio/CSharp/Impl/LanguageService/CSharpCodeStyleConfigureService.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Editor.Implementation.Formatting;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.LanguageServices.CSharp.Options.Formatting;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.LanguageServices.CSharp.LanguageService
+{
+    [ExportWorkspaceService(typeof(ICodeStyleConfigureService)), Shared]
+    internal class CSharpCodeStyleConfigureService : ICodeStyleConfigureService
+    {
+        private CSharpPackage _package;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CSharpCodeStyleConfigureService(SVsServiceProvider serviceProvider)
+        {
+            var shell = (IVsShell)serviceProvider.GetService(typeof(SVsShell));
```

Comment:
I see. I think we want to make it more generic? so that we can use it for any our option pages? or we might already have one?

and i think it would be nice Idea for this to inherit from ForegroundThreadAffinitizedObject and assert on FG

Reply:
updated